core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -68,7 +70,7 @@
    * @return a manifest reader
    */
   public static ManifestReader read(InputFile file) {
```

Comment:
@rdblue, am I correct that using `read(InputFile file)` is not safe only when we update the table schema and use some partition filters (as `ManifestReader` will build a partition spec based on the old schema stored in the metadata of the manifest)?

Reply:
And why is it actually safe to use it without filters? Won't we have the wrong `partition` in `DataFile`?

Reply:
Yes, that's correct. The issue is when the manifest is read using the schema at the time the manifest was written. If a column used by a partition transform is renamed, then expression binding can fail.

It's okay to use it without filters because partition tuples are accessed by position, not by name. So an expression will be bound to the current partition schema.